fh_immuta_utils/scripts/tag_existing_data_sources.py

```diff
@@ -61,18 +61,25 @@ def main(config_file: str, search_text: str, dry_run: bool, debug: bool):
         progress_iterator.set_description(
             desc=f"Tagging ID: {data_source['id']}, Name: {data_source['name']} :"
         )
+        data_source_tags = tagger.get_tags_for_data_source(name=data_source["name"])
+        if data_source_tags:
+            logging.debug(f"Adding data source tags to {data_source['name']}.")
+            if not dry_run:
+                client.tag_data_source(id=data_source["id"], tag_data=data_source_tags)
         dictionary = client.get_data_source_dictionary(id=data_source["id"])
         enriched_columns = tagger.enrich_columns_with_tagging(dictionary.metadata)
         if enriched_columns == dictionary.metadata:
             logging.debug(
-                f"No change for data source: {data_source['name']}. Skipping."
+                f"No change to column tags for data source: {data_source['name']}. Skipping."
```

Comment:
This is super nitpicky, but I'm a fan of lazy evaluation in log entries. I know the syntax for string substitutions in logs in the Python logging stdlib is rather archaic, but I think it's the only way to achieve lazy evaluation and avoid the evaluation of the message if the log level is `> debug`. That being said, it's nitpicky so it's totally up to you!

```suggestion
                "No change to column tags for data source: %s. Skipping.", data_source['name']
```

Reply:
> Looks good to me, @zturechek. I added one minor, nitpicky comments about lazy evaluation of log entries, but otherwise LGTM.

Thanks again @yuvalherziger for taking the time. Re: the lazy evaulation, that's a good knowledge share, thanks. I'll make a note in our backlog to keep on eye on this in the code. For now, I'm going to leave as is since we are only fetching from dicts here.